server/api/budget.mjs

```diff
@@ -0,0 +1,37 @@
+import { mongooseConnect } from "@/lib/mongoose";
+import { Budget } from "@/models/Budget";
+
+export default async function handle(req, res) {
+    const {method} = req; 
+    await mongooseConnect();
+
+
+    if ( method === 'GET') {
+        if (req.query?.id) {
+            res.json(await Budget.findOne({_id:req.query.id}));
+        } else { 
+            res.json(await Budget.find());
+        }
+    }
+
+    if (method === 'POST') {
+        const {budgetName,goalAmount,description,goalDate,status} = req.body;
+        const budgetDoc = await Budget.create({
+            budgetName,goalAmount,description,goalDate,status
+        })
+        res.json(budgetDoc)
+    }
+
+    if (method === 'PUT') {
+        const {budgetName,goalAmount,description,goalDate,status} = req.body;
+        await Budget.updateOne({_id}, {budgetName,goalAmount,description,goalDate,status})
+        res.json(true);
```

Comment:
Usually, in production environments, you actually often return the id of the "item" that is changed